conda_index/utils.py

```diff
@@ -68,4 +68,4 @@ def file_contents_match(pathA, pathB):
     Return True if pathA and pathB have identical contents.
     """
 
-    return filecmp.cmp(pathA, pathB)
+    return filecmp.cmp(pathA, pathB, shallow=False)
```

Comment:
The test for this function failed before doing a non-shallow file comparison here. This might be more expensive, but it seems more thorough, what do you think @dholth?